src/library/scala/math/Ordering.scala

```diff
@@ -224,11 +224,7 @@ object Ordering extends LowPriorityOrderingImplicits {
   implicit object Unit extends UnitOrdering
 
   trait BooleanOrdering extends Ordering[Boolean] {
-    def compare(x: Boolean, y: Boolean) = (x, y) match {
-      case (false, true) => -1
-      case (true, false) => 1
-      case _ => 0
-    }
+    def compare(x: Boolean, y: Boolean) = if(x) { if (y) 0 else 1} else {if (y) -1 else 0}
```

Comment:
>  why is this change needed?

It's not needed. It's just that I saw how inefficient the code was and I didn't resist the temptation to rewrite it. I'm fine with dropping this change if it's not welcome.